multiplexer/abci/abci.go

```diff
@@ -181,6 +181,24 @@ func (m *Multiplexer) VerifyVoteExtension(_ context.Context, req *abci.RequestVe
 	return app.VerifyVoteExtension(req)
 }
 
+func (m *Multiplexer) QuerySequence(ctx context.Context, req *abci.RequestQuerySequence) (*abci.ResponseQuerySequence, error) {
+	// Return noop (sequence 0) for app version <= 4
+	if m.appVersion <= 4 {
+		return &abci.ResponseQuerySequence{Sequence: 0}, nil
+	}
+
+	app, err := m.getApp()
+	if err != nil {
+		return nil, fmt.Errorf("failed to get app for version %d: %w", m.appVersion, err)
+	}
+	resp, err := app.QuerySequence(ctx, req)
+	if err != nil {
+		// If the app doesn't implement QuerySequence, return a noop response
+		return &abci.ResponseQuerySequence{Sequence: 0}, nil
+	}
```

Comment:
[question] here we're silently discarding the error which may make debugging more difficult. Why not propagate the error?

Reply:
this is a good point and I tried changing it, but this causes a lot of tests to fail in testnode as this method is getting called somehwere before

I didn't feel like refactoring testnode just to handle this so I reverted it. happy to open a flup though

https://github.com/celestiaorg/celestia-app/pull/6040/commits/0dbe8fc7765a8eaead2a617c03d225fbb10239d4